openmdao/utils/file_utils.py

```diff
@@ -481,3 +489,117 @@ def _get_outputs_dir(obj=None, *subdirs, mkdir=True):
         dirpath.mkdir(parents=True, exist_ok=True)
 
     return dirpath
+
+
+def _is_openmdao_output_dir(directory):
+    """
+    Check if a directory is an OpenMDAO output directory.
+
+    Parameters
+    ----------
+    directory : str or Path
+        The directory to check.
+
+    Returns
+    -------
+    bool
+        True if the directory is an OpenMDAO output directory, False otherwise.
+    """
+    directory = pathlib.Path(directory)
+    return directory.is_dir() and directory.name.endswith('_out') \
+        and (directory / '.openmdao_out').exists()
+
+
+def _find_openmdao_output_dirs(paths, recurse):
+    """
+    Find all OpenMDAO output directories in the given path.
+
+    Parameters
+    ----------
+    paths : str or Path or Iterable
+        The path to search for OpenMDAO output directories.
+    recurse : bool
+        If True, search recursively.
+
+    Returns
+    -------
+    list
+        A list of OpenMDAO output directories.
+    """
+    if isinstance(paths, (str, pathlib.Path)):
+        paths = [paths]
+    elif not isinstance(paths, Iterable):
+        raise ValueError("The 'paths' parameter must be a string, Path, or an iterable of them.")
+
+    openmdao_dirs = []
+    for path in paths:
+        path = pathlib.Path(path)
+        if not path.is_dir():
+            continue
+
+        for root, dirs, _ in os.walk(path):
+            # Use a copy of the dirs list to avoid modifying it while iterating
+            if _is_openmdao_output_dir(root):
+                openmdao_dirs.append(pathlib.Path(root))
+            for d in dirs[:]:
+                dir_path = pathlib.Path(root) / d
+                if _is_openmdao_output_dir(dir_path):
+                    openmdao_dirs.append(dir_path)
+                    dirs.remove(d)  # Do not recurse into OpenMDAO output directories
+            if not recurse:
+                break
+    return openmdao_dirs
+
+
+def clean_outputs(obj='.', recurse=False, prompt=True, dryrun=False):
+    """
+    Remove output directories created by OpenMDAO.
+
+    A directory is determined to be an OpenMDAO output directory if its name
+    ends in `_out` and it contains the file `.openmdao_out`.
+
+    Parameters
+    ----------
+    obj : Problem or System or Solver or str or Path
+        The problem or system or solver whose output file should be removed.
+    recurse : bool
+        If True, and if obj is a string or Path, recurse into it
+        finding and removing OpenMDAO output directories along the way.
+        This option is ignored if obj is a Problem, System, or Solver.
+    prompt : bool
+        If True, prompt the user to confirm directories to be removed.
```

Comment:
With the "prompt" option, my expectation would be that it would prompt me to delete each one individually instead of all at once. I think that is the only reason i would use this.

Reply:
In the command line usage, prompt is on by default.  The `--force` option will do the deletion without prompting, but otherwise the user will get an "are you sure you want to delete these?" prompt.